matrixprofile/matrixProfile.py

```diff
@@ -151,6 +211,9 @@ def _matrixProfile_stomp(tsA,m,orderClass,distanceProfileFunction,tsB=None):
     order = orderClass(len(tsA)-m+1)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
 
```

Comment:
All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB.

```python
if not is_array_like(tsB):
    tsB = tsA
```